### PR TITLE
Remove Message payload null check

### PR DESCRIPTION
**Pull request by ozangunalp**

### CODE REVIEW

File: api/src/main/java/org/eclipse/microprofile/reactive/messaging/Message.java [hunk 1]
```diff
@@ -56,15 +56,12 @@ static <T> Message<T> of(T payload) {
      * Create a message with the given payload and metadata.
      * The acknowledgement and negative-acknowledgement are immediate.
      *
-     * @param payload The payload, must not be {@code null}.
+     * @param payload The payload.
      * @param metadata The metadata, if {@code null} an empty set of metadata is used.
```
Comment (cescoffier) [inline api/src/main/java/org/eclipse/microprofile/reactive/messaging/Message.java:60]: This code comes from the spec - we cannot change it.

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/eclipse/microprofile/reactive/messaging/Message.java (hunk 1 above)
Comment (ozangunalp) [reply api/src/main/java/org/eclipse/microprofile/reactive/messaging/Message.java:60]: I know but I don't see any non-null requirement in the API code https://github.com/eclipse/microprofile-reactive-messaging/blob/master/api/src/main/java/org/eclipse/microprofile/reactive/messaging/Message.java

Comment (ozangunalp) [reply api/src/main/java/org/eclipse/microprofile/reactive/messaging/Message.java:60]: Some would call this syndrome split-brain :D

File: documentation/src/main/docs/concepts/model.md [hunk 2]
```diff
@@ -72,6 +72,12 @@ Creating `Messages` is done using the {{ javadoc('io.eclipse.microprofile.reacti
 {{ insert('messages/MessageExamples.java', 'creation') }}
 ```
 
+Messages accept `null` as payload.
```
Comment (cescoffier) [inline documentation/src/main/docs/concepts/model.md:75]: I would warn the user that reactive streams do not support `null`, so you cannot consume a `Multi<P>` where the message (Message<P>) has a `null` payload.

**Review by cescoffier — COMMENTED** [review]
Don't we have a section in the kafka doc about tombstone?

**Pull request merged**